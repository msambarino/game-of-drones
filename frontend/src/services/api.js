import axios from 'axios'

axios.defaults.baseURL = process.env.REACT_APP_API_SERVER

export const fetchHighScores = () => {
  return axios.get('/scores')
}

export const registerScore = (player1, player2, winner) => {
  return axios.post('/scores', { player1, player2, winner })
}
