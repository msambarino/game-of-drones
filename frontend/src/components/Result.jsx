import React from 'react'
import PropTypes from 'prop-types'
import { registerScore } from 'services/api'

class Result extends React.Component {
  constructor(props) {
    super(props)

    this.state = {
      loading: false,
      error: null
    }
  }

  componentDidMount() {
    const { player1, player2, winner } = this.props
    this.setState({ loading: true, error: null })

    registerScore(player1, player2, winner)
      .then(response => this.setState({ loading: false }))
      .catch(error => this.setState({ error: true }))
  }

  render() {
    const { winner, handlePlayAgain } = this.props
    const { error } = this.state
    return (
      <div>
        {error ? (
          <div class="notification is-danger">Unable to register score</div>
        ) : null}
        <div className="container">
          <section className="section has-text-centered">
            <h1 className="title">WE HAVE A WINNER!!!</h1>
            <h2 className="subtitle">
              <strong>{winner}</strong> is the new EMPEROR!
            </h2>
            <button className="button is-primary" onClick={handlePlayAgain}>
              Play Again
            </button>
          </section>
        </div>
      </div>
    )
  }
}

Result.propTypes = {
  player1: PropTypes.string.isRequired,
  player2: PropTypes.string.isRequired,
  winner: PropTypes.string.isRequired,
  handlePlayAgain: PropTypes.func.isRequired
}

export default Result
