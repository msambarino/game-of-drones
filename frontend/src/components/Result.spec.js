import React from 'react'
import { shallow, mount, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import Result from './Result'
import { registerScore } from 'services/api'

jest.mock('services/api', () => ({
  registerScore: jest.fn(() => {
    return Promise.resolve({})
  })
}))

test('snapshot test', () => {
  const component = shallow(
    <Result player1="p1" player2="p2" winner="p1" handlePlayAgain={jest.fn()} />
  )
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('should register score when mounted', () => {
  const component = shallow(
    <Result player1="p1" player2="p2" winner="p1" handlePlayAgain={jest.fn()} />
  )
  expect(registerScore).toHaveBeenCalled()
})

test('should call handlePlayAgain when play again button is clicked', () => {
  const playAgainMock = jest.fn()
  const component = shallow(
    <Result
      player1="p1"
      player2="p2"
      winner="p1"
      handlePlayAgain={playAgainMock}
    />
  )

  component.find('button').simulate('click')

  expect(playAgainMock).toHaveBeenCalled()
})
