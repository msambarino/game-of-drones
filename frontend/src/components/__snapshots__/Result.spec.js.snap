// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div>
  <div
    className="container"
  >
    <section
      className="section has-text-centered"
    >
      <h1
        className="title"
      >
        WE HAVE A WINNER!!!
      </h1>
      <h2
        className="subtitle"
      >
        <strong>
          p1
        </strong>
         is the new EMPEROR!
      </h2>
      <button
        className="button is-primary"
        onClick={[Function]}
      >
        Play Again
      </button>
    </section>
  </div>
</div>
`;
